sips/sip-02x/sip-02x-stacks-pay.md

```diff
@@ -0,0 +1,343 @@
+
+- **Warn the User**: Inform the user that the operation type is unrecognized.
+
+- **Provide Safe Defaults**: Default to a standard payment flow if possible.
+
+- **Fail Gracefully**: Prevent unexpected behavior or security risks.
+
+### Parameter Table (summary)
+
+| Operation | token | recipient | amount | description | memo | expiresAt | contractName | functionName | dueDate |
+| --------- | ----- | --------- | ------ | ----------- | ---- | --------- | ------------ | ------------ | ------- |
+| support   | O     | R         | I      | O           | O    | I         | I            | I            | I       |
+| invoice   | R     | R         | R      | O           | O    | O         | I            | I            | O       |
+| mint      | I     | O         | R      | O           | O    | O         | R            | R            | I       |
+| custom    | O\*   | O\*       | O\*    | O\*         | O\*  | O\*       | O\*          | O\*          | O\*     |
+
+```
+R - required
+O - optionalas determined
+I - ignored
+O* - custom links are defined on a per application basis
+```
+
+### Token Types
+
+The `token` parameter indicates the type of token for the payment and **MUST** be one of:
+
+- 'STX': For payments using the native STX token.
+
+- A valid SIP-010 contract address: For payments using SIP-010 compliant fungible tokens.
+
+Note: SIP-010 contract addresses should be in the format `CONTRACT_PRINCIPLE`.`CONTRACT_NAME`, without the associated contract identifier.
+
+### URL Scheme
+
+The Stacks Pay URL scheme **MUST** use the custom protocol `web+stx:`, followed by bech32m encoding of an `operation` and query parameters encoding the payment details with `stx` as the human readable part (hrp) of the encoding. The order of the parameters does not matter.
+
+**Format:**
+
+Format of the url string prior to encoding:
+
+`<operation>?recipient=<recipient>&token=<token>&amount=<amount>[&additional_params]`
+
+For example, here is the unencoded url using 'STX' token:
+
+`invoice?recipient=SP3FBR...&token=STX&amount=1000&description=Payment%20for%20Services`
+
+And here is an example using the SIP-010 Nothing Token:
+
+`invoice?operation=invoice&recipient=SP3FBR...&token=SP32AEEF6WW5Y0NMJ1S8SBSZDAY8R5J32NBZFPKKZ.nope&amount=1000&description=Payment+for+services`
+
+- **`operation`**: Specifies the type of action or transaction. It **MUST** be included and be a string value.
+
+- **Query Parameters**: The payment parameters are appended as URL-encoded query parameters. See specific operation types for which are included for which operations.
+
+Example encoded Stack Pay url:
+
+`stx1wajky2mnw3u8qcte8ghj76twwehkjcm98ahhqetjv96xjmmw845kuan0d93k2fnjv43kjurfv4h8g02n2qe9y4z9xarryv2wxer4zdjzgfd9yd62gar4y46p2sc9gd23xddrjkjgggu5k5jnye6x76m9dc74x4zcyesk6mm4de6r6vfsxqczver9wd3hy6tsw35k7m3a2pshjmt9de6zken0wg4hxetjwe5kxetnyejhsurfwfjhxst585erqv3595cnytfnx92ryve9xdqn2wf9xdqn2w26juk65n`
```

Comment:
This was confusing to read before we got to the encoding section. Maybe save the example for later and focus on defining the `web+stx` format leading into it?

Reply:
It might help to use triple backticks instead of single on the code for readability, e.g.

`stx1wajky2mnw3u8qcte8ghj76twwehkjcm98ahhqetjv96xjmmw845kuan0d93k2fnjv43kjurfv4h8g02n2qe9y4z9xarryv2wxer4zdjzgfd9yd62gar4y46p2sc9gd23xddrjkjgggu5k5jnye6x76m9dc74x4zcyesk6mm4de6r6vfsxqczver9wd3hy6tsw35k7m3a2pshjmt9de6zken0wg4hxetjwe5kxetnyejhsurfwfjhxst585erqv3595cnytfnx92ryve9xdqn2wf9xdqn2w26juk65n`

vs

```
stx1wajky2mnw3u8qcte8ghj76twwehkjcm98ahhqetjv96xjmmw845kuan0d93k2fnjv43kjurfv4h8g02n2qe9y4z9xarryv2wxer4zdjzgfd9yd62gar4y46p2sc9gd23xddrjkjgggu5k5jnye6x76m9dc74x4zcyesk6mm4de6r6vfsxqczver9wd3hy6tsw35k7m3a2pshjmt9de6zken0wg4hxetjwe5kxetnyejhsurfwfjhxst585erqv3595cnytfnx92ryve9xdqn2wf9xdqn2w26juk65n
```